AutoReplayUploader/AutoReplayUploaderPlugin.cpp

```diff
@@ -5,194 +5,341 @@
 }
 
+/**
+* OnUnload event called when the plugin is unloaded by BakkesMod
+*/
 void AutoReplayUploaderPlugin::onUnload()
 {
 }
 
+/**
+* OnGameComplete event called when on Function TAGame.GameEvent_Soccar_TA.EventMatchEnded event when an online game ends.
+* Params:
+*	caller - ServerWraper this event was called from
+*	params - Event parameters
+*	eventName - Event name
+*/
 void AutoReplayUploaderPlugin::OnGameComplete(ServerWrapper caller, void * params, std::string eventName)
 {
-	if (!*uploadToCalculated && !*uploadToBallchasing)
+	if (!*uploadToCalculated && !*uploadToBallchasing) // Bail if we aren't uploading replays
 	{
 		return; //Not uploading replays
 	}
+
+	// Get ReplayDirector
 	ReplayDirectorWrapper replayDirector = caller.GetReplayDirector();
 	if (replayDirector.IsNull())
 	{
 		cvarManager->log("Could not upload replay, director is NULL!");
-		if(*showNotifications) gameWrapper->Toast("Autoreplayuploader", "Error exporting replay! (1)", "default", 3.5f, ToastType_Error);
+#ifdef TOAST
+		if (*showNotifications) gameWrapper->Toast("Autoreplayuploader", "Error exporting replay! (1)", "default", 3.5f, ToastType_Error);
+#endif
 		return;
 	}
+
+	// Get Replay wrapper
 	ReplaySoccarWrapper soccarReplay = replayDirector.GetReplay();
 	if (soccarReplay.memory_address == NULL)
 	{
 		cvarManager->log("Could not upload replay, replay is NULL!");
+#ifdef TOAST
 		if (*showNotifications) gameWrapper->Toast("Autoreplayuploader", "Error exporting replay! (2)", "default", 3.5f, ToastType_Error);
+#endif
 		return;
 	}
-	std::string replayPath = "./bakkesmod/data/autoreplaysave.replay";// cvarManager->getCvar("cl_autoreplayupload_filepath").getStringValue(); //"./bakkesmod/data/autoreplaysave.replay";//
+
+	// If we have a template for the replay name then set the replay name based off that template
+	auto replayNameTemplate = cvarManager->getCvar(CVAR_REPLAY_NAME_TEMPLATE).getStringValue();
+	if (!replayNameTemplate.empty())
+	{
+		SetReplayName(caller, soccarReplay, replayNameTemplate);
+	}
+
+	// Determine the replay path and if it already exists remove it
+	std::string replayPath = "./bakkesmod/data/autoreplaysave.replay";// cvarManager->getCvar(CVAR_REPLAY_UPLOAD_PATH).getStringValue();
 	if (file_exists(replayPath))
 	{
 		cvarManager->log("Removing existing file: " + replayPath);
 		remove(replayPath.c_str());
 	}
-	cvarManager->log("Exporting replay to " + replayPath);
-	soccarReplay.ExportReplay(replayPath);
-	cvarManager->log("Replay exported!");
-	if (*uploadToCalculated) 
+
+	// Export Replay
+	soccarReplay.ExportReplay(replayPath); // %HOMEPATH%\Documents\My Games\Rocket League\TAGame\Demos
+	cvarManager->log("Exported replay to " + replayPath);
+
+	// Upload replay
+	if (*uploadToCalculated)
 	{
-		UploadReplayToEndpoint(replayPath, CALCULATED_ENDPOINT_DEFAULT, "replays", "", "calculated.gg");
+		UploadReplayToEndpoint(replayPath, CALCULATED);
```

Comment:
Rather than have all the parameters passed into this function I figured it would be more maintainable in the future if the function new what to do different for each different endpoint.  Then if we add another endpoint we don't have to extend the method signature and do a bunch hacky logic.

Reply:
Agreed